### ISSUE TITLE
Separate case blocks and other blocks for whitespaces

### ISSUE THREAD

**Issue by bombsimon** [bombsimon/wsl#25]
Blocks cannot start or end with a whitespace. I think this is a very reasonable rule except for maybe the possibility to end a `case` block with a whitespace. I'm open to the idea to at least have this configurable and maybe even as the default since this can actually improve readability and mark where the next case starts. For this to be reasonable it's important to ensure this **does not** apply to the last `case` since that would actually mena the ending of the `switch` block and not the `case` block.

```go
[token] {
    // Can't be a whitespace here
    fmt.Println("...")
    // Can't be a whitespace here
}

switch {
// Can't be a whitespace here
case 1:
    // Can't be a whitespace here
    fmt.Println("...")
    // Can't be a whitespace here - should it be possible?
case 2:
    // Can't be a whitespace here
    fmt.Println("...")
    // Can't be a whitespace here - should it be possible?
case 3:
    // Can't be a whitespace here
    fmt.Println("...")
// Can't be a whitespace here
}
```

**Comment by hollowaykeanho** [comment]
I'm getting a weird report about "end with whitespace".

```
func (s *testENDECScenario) prepareENDEC(e *ENDEC) (result string) {           
	switch {                                                               
	case s.Switches[convertToBigFloat]:                                    
		e.fType = floatBig                                             
		e.fb = s.prepareBigFloat()                                     
	case s.Switches[usePositiveFloat32] ||                                 
		s.Switches[useZeroFloat32] ||                                  
		s.Switches[useNegativeFloat32] ||                              
		s.Switches[useNegativeExponentFloat32]:                        
		e.f32 = s.prepareFloat32()                                     
		e.fType = float32bit                                           
	case s.Switches[usePositiveFloat64] ||                                 
		s.Switches[useZeroFloat64] ||                                  
		s.Switches[useNegativeFloat64]:                                
		e.fType = float64bit                                           
		e.f64 = s.prepareFloat64()                                     
	}                                                                      
                                                                               
	return s.prepareStringResult()                                         
}       
```

for 3rd case. No spotting of ending whitespace. How to fix it?

-----

Suggest configurable but default to no space so that multi-line blocks can be neatly arranged like:

```
func (s *testENDECScenario) prepareENDEC(e *ENDEC) (result string) {           
	switch {                                                               
	case s.Switches[convertToBigFloat]:                                    
		e.fType = floatBig                                             
		e.fb = s.prepareBigFloat()       
                              
	case s.Switches[usePositiveFloat32] ||                                 
		s.Switches[useZeroFloat32] ||                                  
		s.Switches[useNegativeFloat32] ||                              
		s.Switches[useNegativeExponentFloat32]:                        
		e.f32 = s.prepareFloat32()                                     
		e.fType = float32bit              
                             
	case s.Switches[usePositiveFloat64] ||                                 
		s.Switches[useZeroFloat64] ||                                  
		s.Switches[useNegativeFloat64]:                                
		e.fType = float64bit                                           
		e.f64 = s.prepareFloat64()                                     
	}                                                                      
                                                                               
	return s.prepareStringResult()                                         
}       
```
```

**Comment by bombsimon** [comment]
@hollowaykeanho Thanks for your report! Definitely a bug, I've opened #29 and will try to look into that asap.

**Comment by bombsimon** [comment]
@hollowaykeanho The bug in #29 is now fixed, merged and in release [v1.2.3](https://github.com/bombsimon/wsl/releases/tag/v1.2.3). I'll try to look into this issue before next week and implement it.

**Comment by hollowaykeanho** [comment]
@bombsimon noted with thanks!
